File: codingTest/baekjoon/javascript/9037.js
const rl = require("readline").createInterface({ input: process.stdin, output: process.stdout });

const check = arr => {
  for (const i in arr) if (arr[i] % 2 == 1) arr[i] += 1;
  return new Set(arr).size == 1;
};

const input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  let T = +input.shift();
  while (T--) {
    const N = +input.shift();
    const arr = input
      .shift()
      .split(" ")
      .map(e => +e);
    const half = new Array(N);
    let cnt = 0;
    while (!check(arr)) {
      for (const i in arr) half[i] = arr[i] /= 2;
      arr[0] += half[N - 1];
      for (let i = 0; i < N - 1; i++) arr[i + 1] += half[i];
      cnt++;
    }
    console.log(cnt);
  }
});

File: codingTest/baekjoon/python/9037.py
# def isSame(arr, N):
#     for i in range(N - 1):
#         if arr[i] != arr[i + 1]:
#             return False
#     return True

def isSame(arr):
    if len(set(arr)) == 1:
        return True
    return False


T = int(input())
while T:
    N = int(input())
    arrC = list(map(int, input().split()))

    count = 0
    while True:
        for i in range(N):
            if arrC[i] % 2 != 0:
                arrC[i] += 1
        if isSame(arrC):
            break
        arrPlus = list(map(lambda x: x // 2, arrC))
        for i in range(N):
            arrC[i] -= arrPlus[i]
        arrC[0] += arrPlus[N - 1]
        for i in range(N - 1):
            arrC[i + 1] += arrPlus[i]
        count += 1
    print(count)
    T -= 1
